### ISSUE TITLE
change all_trees_exist to all_blobs_exist

### ISSUE THREAD

**Issue by nikita-skobov** [nikita-skobov/monorepo-git-tools#15]
I noticed an issue where a tree can potentially be different on a `--topbase` because of excluded files that werent originally excluded, but now are. its difficult to describe a reproducible example, but basically my original idea of checking if the all root trees of a commit exist to determine if a commit should be included in the topbase or not, was wrong. it is insufficient to only check for the root trees. We should actually check for all **blobs** instead.

basically do this:

- on all_blobs_exist:
    - iterate recursively through the current tree, and find all end blob nodes, put into a vector
- iterate over the vector, and for each blob, check if it exists SOMEWHERE in the list of commits (this involves iterating over every commit, for every blob... might be slow)
- if a single blob doesnt exist in the list of upstream commits, return false. otherwise return true to indicate that every single blob exists
